### DISCUSSION TITLE
SmartRF Flash Programmer 2 download not available

### DISCUSSION THREAD

**Discussion by oh2xx** [Koenkk/zigbee2mqtt#10476] [General]
Hi,

TI support has been offline ("Sorry! We are currently doing site maintenance.") now for a few days and I am not able register to download the flashing program. Is there someone who could share his/her copy of the SW with me? I could not find any alternative (trustful) download site either.

I have a new Sonoff Zigbee 3.0 and need to flash it as I am going to change the IEEE address.

**Comment by kirelam** [comment]
I can send flash programmer to you if you like? Can you advice me on how to change the IEEE adress? I tried many things, but nothing worked.

**Reply by oh2xx** [reply]
Please upload it here https://www.dropbox.com/request/EYRTrl5nzp1rKnIZHH3p thank you!

**Reply by oh2xx** [reply]
> 

I am following these instructions https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html to change the address. Never done it before though.

**Comment by oh2xx** [comment]
I have not received the SW from @kirelam yet so if there is anyone else who could help it would be very much appreciated.. I keep the upload link alive for now.

**Comment by kirelam** [comment]
I Will try to provider the software, but this Will be in a few hours, or
tomorrow.

Op za 1 jan. 2022 17:52 schreef oh2xx ***@***.***>:

**Comment by BlueScreenTT** [comment]
go with Linux and the python script !!

**Reply by oh2xx** [reply]
Tried uartlog.py in windows but got an error message: 

C:\Users\oh2xx\uartLog>python uartLog.py
Traceback (most recent call last):
  File "C:\Users\Kari\OneDrive\Tiedostot\uartLog\uartLog.py", line 10, in <module>
    import serial
ModuleNotFoundError: No module named 'serial'

Tried to google a solution for that one but could not find any.

But that is just to enter Bootloader. If I do it by pressing the button where is the Python script to do the flashing part? Can google that myself too but if you have it at hand please share. I will be back at this tomorrow.

https://sonoff.tech/wp-content/uploads/2021/09/Zigbee-3.0-USB-dongle-plus-firmware-flashing-1-1.docx does not say anything about a python script to flash or does it?

**Comment by BlueScreenTT** [comment]
Check youtube digiblurdiy. He got  a good guide for Windows and Linux

https://youtu.be/4S_c_m6z-RY

//Thomas

**Reply by oh2xx** [reply]
I followed this path as I did not receive any software from @kirelam. Spent many hours and learned a lot. It kind of works now but I do get a few errors still, like:

Zigbee2MQTT:error 2022-01-02 23:04:08: Publish 'set' 'state' to 'rappuset_valo' failed: 'Error: Command 0x84fd27fffecf2afb/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'No network route' (205))'

It seems that the Ikea light bulbs are not receiving my commands but TuYa thermometers report temperature and humidity ok.

It would have been much easier to just change the adapter without any IEEE address changes and do the pairing, but wanted to learn this process for the future if my Zigbee network expands considerably. I still may have to do the pairing again if the Ikea light bulbs will not connect.

**Reply by oh2xx** [reply]
Ok, it just took a few hours for the coordinator to figure it all out. Now everything seems to work just fine. 

My key learnings were:
- cc2538-bsl is the easiest way to change the IEEE address and to flash at the same time as required https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html - try it first!
- watch the https://youtu.be/4S_c_m6z-RY video carefully ;-)
- With Raspberry Pi OS you need to use "python3" command not "python" for the cc2538-bsl script to work
- remember to change the /dev/ttyXXX in configuration.yaml after changing the adapter
